### PR TITLE
Fix DateTimeOffsetJsonConverter

### LINKED ISSUE #828: Skipped Test: JsonDeserializationBugTests.ShouldPreserveTimezoneInformation
Re-enable the `JsonDeserializationBugTests.ShouldPreserveTimezoneInformation` test.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1185) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixed #828

### CODE REVIEW

File: src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        public void ShouldSetNullForExcludedProperty()
+        {
+            var originalDateTimeOffset = new NullableTestClass();
+            var serializedDateTimeOffset = JsonHelper.Serialize(originalDateTimeOffset);
+            var serializedJToken = JToken.Parse(serializedDateTimeOffset);
+            serializedJToken["DateTimeOffset"].Should().BeNull();
+
+            var deserializedDateTimeOffset = JsonHelper.Deserialize<NullableTestClass>(serializedDateTimeOffset);
+            deserializedDateTimeOffset.DateTimeOffset.Should().BeNull();
+        }
```
Comment (marto83) [inline src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs:97]: You are missing a test with NullableTestClass with a datetime offset value.

File: src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        [Fact]
+        public void ShouldConvertDateTimeOffsetToMilliseconds()
+        {
+            var originalDateTimeOffset = new TestClass(new DateTimeOffset(2014, 1, 1, 0, 0, 0, TimeSpan.Zero));
+            var serializedDateTimeOffset = JsonHelper.Serialize(originalDateTimeOffset);
+            var serializedJToken = JToken.Parse(serializedDateTimeOffset);
+            serializedJToken["DateTimeOffset"].Should().NotBeNull(); // check that json property/key exists
+            serializedJToken["DateTimeOffset"].Type.Should().Be(JTokenType.Integer); // check type of the property
+            serializedJToken["DateTimeOffset"].Value<long>().Should().Be(1388534400000);
+
```
Comment (marto83) [inline src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs:30]: Can you fix the white space warnings?

**Review by marto83 — COMMENTED** [review]
Looks good, just a couple of comments

File: src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs (hunk 2 above)
Comment (sacOO7) [reply src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs:29]: Done

File: src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs (hunk 1 above)
Comment (sacOO7) [reply src/IO.Ably.Tests.Shared/CustomSerializers/DateTimeOffsetConverterTests.cs:91]: Done

**Review by tomkirbygreen — APPROVED** [review]
Thank you Sachin! Hope you have a peaceful weekend.

**Pull request merged**